src/pages/Dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { IApiResponse, IResultData, Istats, Result } from 'utils/types';
+import { getResult, deleteResult } from 'api/result';
+import { CONSTANTS } from 'utils/constant';
+import Modal from 'components/Modal/Modal';
+import showToast from 'utils/toastMessage';
+import DashboardHeader from 'components/DashboardHeader';
+import SummaryCards from 'components/SummaryCards';
+import Table from 'components/Table';
+import 'pages/Dashboard/style.css';
+
+function Dashboard() {
+  const navigate = useNavigate();
+  const [students, setStudents] = useState<IResultData[]>([]);
+  const [dropdownVisible, setDropdownVisible] = useState(false);
+  const [selectedStudent, setSelectedStudent] = useState<IResultData | null>(null);
+  const [showModal, setShowModal] = useState(false);
+  const [stats, setStats] = useState<Istats | undefined>(undefined);
+
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response: IApiResponse = await getResult();
+        if (response.success) {
+          setStudents(response.data);
+          calculateStats(response.data);
+        }
+      } catch (error: IApiResponse | any) {
+        showToast(error);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const calculateStats = (resultData: IResultData[]) => {
+    if (resultData.length === 0) return;  
+    const subjectPassCount: Result = {};
+    const subjectFailCount: Result = {};
+  
+    let highestGrade = 'F';
+    let lowestGrade = 'A+';
+  
+    resultData.forEach(data => {
+      const { subjectName } = data.subjectId;
+      const { grade } = data;
+  
+      if (!subjectPassCount[subjectName]) subjectPassCount[subjectName] = 0;
+      if (!subjectFailCount[subjectName]) subjectFailCount[subjectName] = 0;
+  
+      if (grade === 'F') {
```

Comment:
don't do these calculation inside component, purpose of component should be displaying data and this data should be coming as prop

Reply:
implement selectors